lib/DefaultResolver.php

```diff
@@ -214,6 +218,8 @@ private function doResolve($name, array $types, $options) {
             $this->config = (yield \Amp\resolve($this->loadResolvConf()));
         }
 
+        $config = $this->loadEnvConfig();
```

Comment:
Ironically $_ENV and setenv are not synchronized even. People are relying on implementation details if they use setenv(). (as you use getenv() and not $_ENV).

If you want to allow that, provide an explicit API to change the global configuration.